gdk/commands/Command.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from abc import abstractmethod
+
+import gdk
+from gdk.common.exceptions.CommandError import ConflictingArgumentsError
+
+
+class Command:
+    def __init__(self, arguments, name) -> None:
+        self.arguments = arguments
+        self.name = name
+        logging.debug("Checking if arguments in the command conflict.")
+        self.check_if_arguments_conflict()
+
+    def check_if_arguments_conflict(self):
+        """
+        Checks if the command namespace provided to the parser has conflicting arguments
+
+        Returns
+        -------
+          (bool) Returns True if the command arguments conflict. Else False.
+        """
+
+        _non_conflicting_args_map = self._non_conflicting_args_map()
+        if _non_conflicting_args_map:
+            self._identify_conflicting_args_in_command(_non_conflicting_args_map)
+
+    def _identify_conflicting_args_in_command(self, _non_conflicting_args_map):
+        """
+        Checks if the command namespace provided to the parser has conflicting arguments by using the conflicting
+        arguments provided in the cli model file.
+
+        Parameters
+        ----------
+          _non_conflicting_args_map(dict): A dictionary object that's formed with argument as a key and
+                              a set of its non-conflicting args as value.
+
+        Returns
+        -------
+          (bool) Returns True if the command arguments conflict. Else False.
+        """
+        command_arg_keys = self._arguments_list(_non_conflicting_args_map)
+        for i in range(len(command_arg_keys)):
```

Comment:
We're using `argparse` as the tool builder. Using  `add_mutually_exclusive_group` of argparse, we cannot add more than one set of conflicting groups. There used to be a library for doing this https://pypi.org/project/ConflictsOptionParser/ but because it is not actively maintained, I wrote our own.